code/modules/clothing/chameleon/_chameleon_outfit.dm

```diff
@@ -0,0 +1,121 @@
+/datum/action/chameleon_outfit
+	name = "Select Chameleon Outfit"
+	desc = "LMB: Select a job to update all of your chameleon items to.<br>\
+		RMB: Save your current chameleon setup as a custom outfit."
+	button_icon_state = "chameleon_outfit"
+	check_flags = AB_CHECK_CONSCIOUS|AB_CHECK_INCAPACITATED|AB_CHECK_HANDS_BLOCKED
+
+	/// Determines when we're in use
+	var/applying = FALSE
+	/// Cached assoc list of job outfit datums by their names that we can select
+	/// If you intend on editing this, ensure you are copying it first
+	var/list/outfit_options
+	/// Assoc list of custom outfit names ("Custom outfit 1", "Custom outfit 2", etc) to list of all item typepaths saved in that outfit
+	var/list/custom_outfits
+	/// Cap on how many custom outfits we can save
+	var/max_custom_oufits = 10
+	/// Used to generate the next custom outfit name, also to track which next to override if we've passed the limit
+	var/next_custom_outfit = 1
+
+/datum/action/chameleon_outfit/New()
+	. = ..()
+	outfit_options = get_initial_outfits()
+
+/datum/action/chameleon_outfit/proc/get_initial_outfits()
+	var/static/list/standard_outfit_options
+	if(!standard_outfit_options)
+		standard_outfit_options = list()
+		for(var/datum/outfit/found_outfit as anything in subtypesof(/datum/outfit/job))
+			standard_outfit_options[initial(found_outfit.name)] = new found_outfit()
+		sortTim(standard_outfit_options, GLOBAL_PROC_REF(cmp_text_asc))
+
+	return standard_outfit_options
+
+/datum/action/chameleon_outfit/Trigger(trigger_flags)
+	. = ..()
+	if(!.)
+		return
+	if(usr != owner) // i guess
+		return
+	if(applying)
+		return
```

Comment:
```suggestion
	if(!. || applying)
		return
	if(usr != owner) // i guess
		return
```

you can also condense the usr/owner check into that line but there's no real pressing urge to, just mentioning this one at least so we don't have to do that equals check if it's going to be returned anyways